### ISSUE TITLE
hide method is not working in android

### ISSUE THREAD

**Issue by binita12345** [EddyVerbruggen/cordova-plugin-safariviewcontroller#62]
Hi i am trying to hide my safariViewController using safariViewController.hide().
but it is throwing error: "Invalid action".
Help me.

**Comment by EddyVerbruggen** [comment]
Share your code please.

**Comment by binita12345** [comment]
Hi Eddy,
i am working on ionic2.
here is my code


 this.safariViewController.isAvailable()
.then((available: boolean) => {

if (available) {
console.log('safari view test start');
this.safariViewController.show({
url: data.link,
       hidden: false,
       animated: false,
       transition: 'curl',
       enterReaderModeIfAvailable: true,
       tintColor: '#ff0000'
   })
   .then((result: any) => {
    console.log("result*****: "+JSON.stringify(result));
    if(result.event === 'opened') {
    console.log('Opened');
    console.log("data");
    console.log("data: "+JSON.stringify(data));
    setTimeout(function() {
      console.log('TESTTTT');
this.safariViewController.hide();
}, 5000);


    }
    else if(result.event === 'loaded'){
    console.log('Loaded');

    }
    else if(result.event === 'closed') console.log('Closed');
   },
   (error: any) => console.error(error)
   );
   console.log('safari view test');


} else {
// use fallback browser, example InAppBrowser
}

});

**Comment by EddyVerbruggen** [comment]
Could it be the context of `this` is lost after the timeout? So `this.safariViewController.hide` will no longer work.

Looks like you're using TS, so perhaps try this instead, as arrow functions keep track of `this` for you:

```js
setTimeout(() => {
  console.log('TESTTTT');
  this.safariViewController.hide();
}, 5000);
```

If that works you can donate me some 🍺  on my Paypal :)

**Comment by binita12345** [comment]
nope context of "this" ll not be lost throughout my function.
same is working fine for me in IOS but it throws error in android.

**Comment by EddyVerbruggen** [comment]
You're right, `hide` wasn't implemented on Android. I didn't provide the Android implementation so I don't know why not. Perhaps it's not possible even.

Turning this into a feature request.

**Comment by binita12345** [comment]
okay thank you Eddy for your response :)

**Comment by Dimigergo** [comment]
Hello!

What is the status of this issue? Is it possible that the "hide" method in Android will be implemented in the next release?

Thanks in advance!

**Comment by EddyVerbruggen** [comment]
You're welcome to give it a shot if you need it. I personally don't, so..

**Comment by semiaddict** [comment]
@Dimigergo, @binita12345 did any of you manage to implement this for Android ?
If so, please share.

For my purpose (an oauth authentication), applying the changes described in https://stackoverflow.com/questions/42530178/auto-close-chrome-custom-tab#42530510 closes Android's custom tab after redirecting to the app, which is exactly what I needed.

The changes done are:

1. add the following line to config.xml:
`<preference name="AndroidLaunchMode" value="singleTop" />`

2. add the following lines at the begining of startCustomTabActivity in plugins/cordova-plugin-safariviewcontroller/src/android/ChromeCustomTabPlugin.java:
```
intent.intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
intent.intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```

Refer to https://bugs.chromium.org/p/chromium/issues/detail?id=543542 , https://bugs.chromium.org/p/chromium/issues/detail?id=545446 , and 
https://stackoverflow.com/questions/42530178/auto-close-chrome-custom-tab for more information.

**Comment by semiaddict** [comment]
Actually, applying the first change, by adding the preference to config.xml seems to be enough to automatically close the tab when redirecting to the app.

**Comment by PowerMogli** [comment]
Is it possible to implement just a dummy action for android so the exception does not occur anymore.
